packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -212,18 +212,30 @@ class _TextEditingValueAccumulator {
   }
 }
 
-/// A [TextInputFormatter] that prevents the insertion of characters
-/// matching (or not matching) a particular pattern.
+/// A [TextInputFormatter] that prevents the insertion of characters matching
+/// (or not matching) a particular pattern, by replacing the characters with the
+/// given [replacementString].
 ///
 /// Instances of filtered characters found in the new [TextEditingValue]s
-/// will be replaced with the [replacementString] which defaults to the empty
+/// will be replaced by the [replacementString] which defaults to the empty
 /// string.
 ///
-/// Since this formatter only removes characters from the text, it attempts to
-/// preserve the existing [TextEditingValue.selection] to values it would now
-/// fall at with the removed characters.
+/// A [FilteringTextInputFormatter] also tries to preserve the existing
+/// [TextEditingValue.selection] and [TextEditingValue.composing], and adjusts
```

Comment:
nit: maybe add a "region" after "[TextEditingValue.composing]" to make this read better?